ui/components/app/signature-request-siwe/signature-request-siwe.js

```diff
@@ -51,8 +51,9 @@ export default function SignatureRequestSIWE({
     let isSIWEDomainValid = false;
 
     if (origin) {
-      const { host } = new URL(origin);
-      isSIWEDomainValid = parsedMessage.domain === host;
+      const { hostname, host } = new URL(origin);
+      isSIWEDomainValid =
+        parsedMessage.domain === hostname || parsedMessage.domain === host;
```

Comment:
This loosening of origin matching doesn't seem to be limited to localhost in any way. As a result, any two origins (in the browser's sameorigin policy sense) that share the domain but not port&protocol will be considered a match. 

Are we not worried about changing that? 
Shared hosting services exist, where two different apps/customers get the same IP or domain but different port. I know it's a weird example, but it's a practical example of origin including a port.

If we consider it an extension of sameorigin policy, we should not fix localhost issues this way. 

(also, sorry this is coming in so late, but I was only made aware of this PR a few minutes ago and I don't even have all the context - I might be 100% wrong)

Reply:
Would this work too? There's no security concerns I see if done that way:  
```suggestion
        (hostname === 'localhost' && parsedMessage.domain === hostname) || parsedMessage.domain === host;
```